packages/core/src/components/popover/popover.tsx

```diff
@@ -717,11 +719,7 @@ export class Popover<
         if (!shouldIgnoreClick) {
             // ensure click did not originate from within inline popover before closing
             if (!this.props.disabled && !this.isElementInPopover(e.target as HTMLElement)) {
-                if (this.props.isOpen == null) {
-                    this.setState(prevState => ({ isOpen: !prevState.isOpen }));
-                } else {
-                    this.setOpenState(!this.props.isOpen, e);
-                }
```

Comment:
I noticed this code block was redundant - `setOpenState()` already has logic to distinguish between the two cases in this if/else statement.

Reply:
nevermind -- this broke some popover closing behavior, will leave this code in place